### PR TITLE
7041 - Replacing Streamfield blocks Stylings

### LINKED ISSUE #7041: 3 - Replacing Streamfield blocks Stylings

### PR DESCRIPTION
**Pull request by fessehaye**
Closes #7041

Goal was replace most of the streamfield css classes with tailwind equivalents and remove custom css used that isn't affecting the rich text fields!
 
Bootstrap and tailwind uses different spacing increments so it is likely there might have been some mistakes on my side when I was switching.
There are a lot of files that were affected so take your time reviewing the changes!

## Checklist
* Check the style-guide for visual regressions
* Check Percy for visual regressions
* Check review app for visual regressions

### CODE REVIEW

File: source/js/components/join/join.jsx [hunk 1]
```diff
@@ -482,7 +482,7 @@ class JoinUs extends Component {
   renderSubmitButton() {
     let classnames = classNames("btn btn-primary", {
       "w-100": !this.isFlowForm(),
-      "flex-1 mr-3": this.isFlowForm(),
+      "tw-flex-1 mr-3": this.isFlowForm(),
```
Comment (tbrlpld) [inline source/js/components/join/join.jsx:485]:
```suggestion
      "tw-flex-1 tw-mr-3": this.isFlowForm(),
```

**Review by tbrlpld — CHANGES_REQUESTED** [review]
Nice work @fessehaye. Only a few places where I think some classes were missed, which is only a tiny fraction of this huge update 👍

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
-            <div class="col-sm-12 {% if count > 2 %} col-lg-4 col-md-6 {% else %} col-md-6 {% endif %} mb-4">
-              <div class="card-regular h-100 d-flex flex-column ">
+            <div class="tw-px-4 small:tw-w-full {% if count > 2 %} large:tw-w-1/3 medium:tw-w-1/2 {% else %} medium:tw-w-1/2 {% endif %} tw-mb-5">
+              <div class="card-regular tw-h-full tw-flex tw-flex-col">
                 <img src="{{ img.url }}" alt="{{ img.alt_text }}" />
             
-                <div class="d-flex flex-1">
-                  <div class="key-item mx-2 mx-md-3 p-3 w-100 d-flex flex-column">
-                    <h3 class="h3-heading mb-2">{{ card.title }}</h3>
+                <div class="d-flex tw-flex-1">
```
[the remark this answers is not in the document]
Comment (tbrlpld) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html:13]:
```suggestion
                <div class="tw-flex tw-flex-1">
```

**Review by tbrlpld — APPROVED** [review]

**Review by tbrlpld — CHANGES_REQUESTED** [review]
There are still some visual differences on Percy: https://percy.io/Mozilla-Foundation/foundation.mozilla.org/builds/15812634/changed/892390096?browser=chrome&browser_ids=19%2C20&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280

Looks like the negative margins are still not quite right.

**Review by danielfmiranda — COMMENTED** [review]
Hi Simon! Thanks so much for taking this work on as it did seem to be quite a lift! I think this is definitely going to help our efforts in moving from bootstrap to tailwind. 

After taking a look at the review app and the tests, it appears that I am experiencing the same thing as Tibor regarding the negative margins on the card. 

Once this is fixed let me know and I will be happy to retest!

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html [hunk 3]
```diff
@@ -2,20 +2,20 @@
...
-              <div class="card-regular h-100 d-flex flex-column ">
+            <div class="tw-px-4 small:tw-w-full {% if count > 2 %} large:tw-w-1/3 medium:tw-w-1/2 {% else %} medium:tw-w-1/2 {% endif %} tw-mb-5">
+              <div class="card-regular tw-h-full tw-flex tw-flex-col">
                 <img src="{{ img.url }}" alt="{{ img.alt_text }}" />
             
-                <div class="d-flex flex-1">
-                  <div class="key-item mx-2 mx-md-3 p-3 w-100 d-flex flex-column">
-                    <h3 class="h3-heading mb-2">{{ card.title }}</h3>
+                <div class="tw-flex tw-flex-1">
+                  <div class="tw-border-b-4 tw-border-gray-05 tw--mt-5 tw-bg-white tw-relative tw-mx-2 medium:tw-mx-4 tw-p-4 tw-w-full tw-flex tw-flex-col">
```
Comment (tbrlpld) [inline network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html:14]: **Question:** Wouldn't this be `-tw-mt-5` now that the update is in? Not sure what our usual approach here is (mainly for readability).

**Review by tbrlpld — APPROVED** [review]
@fessehaye This looks good to me 👍 Just left one question.

**Review by tbrlpld — CHANGES_REQUESTED** [review]
Whoops. Sorry I should have checked Percy first. It looks like there is something pulling an  image on [the fixed blog post](https://percy.io/Mozilla-Foundation/foundation.mozilla.org/builds/15968012/changed/900148686?browser=chrome&browser_ids=19%2C20&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280) to the left:

Screenshot 2022-02-23 at 06 19 11


This seems to happen on [other pages](https://percy.io/Mozilla-Foundation/foundation.mozilla.org/builds/15968012/changed/900148527?browser=chrome&browser_ids=19%2C20&subcategories=unreviewed%2Cchanges_requested&viewLayout=side-by-side&viewMode=new&width=1280&widths=375%2C1280) too.

File: network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html (hunk 3 above)
Comment (fessehaye) [reply network-api/networkapi/wagtailpages/templates/wagtailpages/blocks/card_grid_block.html:14]: Either or works now and intellsense hover will show the property on vscode

**Review by tbrlpld — APPROVED** [review]
Thanks @fessehaye this is great

**Review by danielfmiranda — APPROVED** [review]
Looks good to me Simon, and again, awesome job! 👍 approved

**Pull request merged**